### ISSUE TITLE
Indentscope: Add option to ignore small scope

### ISSUE THREAD

**Issue by xzbdmw** [echasnovski/mini.nvim#815]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)

### Module(s)

mini.indentscope

### Description

Thanks for your plugin!
I often find it is disturb to draw animation in small if-else which only contain one or two lines during cursor movement, and I want to show the indent which is one lever "bigger" than small scopes, by that case, goto_top is more useful.

**Comment by echasnovski** [comment]
Thanks for the issue!

First instinct is that this is a bit too much configuration for what I perceive as not a huge issue. But as this might fit together with possible solution of #568 (via introducing upper bound of when to draw or stop drawing indentscope line), I'll think about it.

**Comment by niveK77pur** [comment]
I want to second this! A very small scope of 1-3 lines is often not all that interesting to visualize. I was hoping to find an option where I can set the minimum number of lines that should cause the indent to be shown in a nested scope.

Also, this is a really sweet plugin! Thanks a lot for all the work so far!

**Comment by echasnovski** [comment]
As I don't think having a solution to #568 via upper bound is a good approach, I am afraid I am going to close this issue. This indeed feels like too much configuration for something that I don't feel like a crucial and/or universally useful feature. Thanks for the suggestion, though!

**Comment by echasnovski** [comment]
@xzbdmw, I finally decided to resolve #568 and the only good way to do it was indeed with introducing limit on scope computation. I decided to go with strict line count restriction (as in 'mini.ai' and 'mini.surround') which makes scope marked as "incomplete".

As scopes with incomplete computations usually contain information which is actively harmful for drawing (as it usually covers text), those should not be drawn. To achieve that, I decided to go a bit more generic route and introduce `config.draw.predicate` which can be used to decide whether scope needs to be drawn. This also should resolve this issue with config which looks something like this:

```lua
local draw_predicate = function(scope)
  return not scope.body.is_incomplete and (scope.body.bottom - scope.body.top + 1) >= 3
end
require('mini.indentscope').setup({ draw = { predicate = draw_predicate } })
```
